File: Administrativo/spring-erp-servicio/net/royal/spring/core/servicio/ZonapostalServicio.cs
using System;
using System.Collections.Generic;
using System.Text;

using net.royal.spring.framework.web.servicio;
using net.royal.spring.core.dominio;
using net.royal.spring.core.dominio.filtro;

namespace net.royal.spring.core.servicio
{

public interface ZonapostalServicio : GenericoServicio {
        List<Zonapostal> listarTodos();
        List<Zonapostal> listar(FiltroZonaPostal filtro);
        List<Zonapostal> listarActivosPorProvincia(String idDepartamento, String idProvincia);

        String obtenerNombrePorId(String departamento,String provincia,String distrito);
    }
}

File: Administrativo/spring-erp-web/SpringNg/src/app/erp_module/shared/zonapostal/servicio/ZonapostalServicio.ts
import { Component, Inject, Injectable } from '@angular/core';
import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
import { Header } from 'primeng/primeng';
import { StringifyOptions } from 'querystring';
import { Zonapostal } from '../dominio/Zonapostal';

@Injectable()
export class ZonapostalServicio {

    private rutaServicio: string;
    private http: HttpClient;
    private baseUrl: string;

    constructor(http: HttpClient, @Inject('BASE_URL') baseUrl: string) {
        this.http = http;
        this.baseUrl = baseUrl;
        this.rutaServicio = this.baseUrl + 'api/spring/core/Zonapostal/';
    }

    public listarActivosPorProvincia(idDepartamento: string, idProvincia: string): Promise<Zonapostal[]> {

        const params = new HttpParams()
            .set('idDepartamento', idDepartamento)
            .set('idProvincia', idProvincia);

        return this.http.get(this.rutaServicio + 'listarActivosPorProvincia', { params })
            .toPromise()
            .then(response => {
                return response as Zonapostal[];
            })
            .catch(error => new Array());
    }

    public obtenerNombrePorId(departamento: string, provincia: string, distrito: string): Promise<string> {

        const params = new HttpParams()
            .set('departamento', departamento)
            .set('provincia', provincia)
            .set('distrito', distrito);

        return this.http.get(this.rutaServicio + 'obtenerNombrePorId', { params })
            .toPromise()
            .then(response => {
                return (response as any).nombre;
            })
            .catch(error => '');
    }



}
